src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -212,16 +212,21 @@ private void createBranchSensitivityValue(LfSensitivityFactor factor, DenseMatri
             sensiValue = factor.getBaseSensitivityValue();
             flowValue = factor.getFunctionReference();
             for (ComputedContingencyElement contingencyElement : contingencyElements) {
-                if (contingencyElement.getElement().getId().equals(functionBranchId)
-                        || contingencyElement.getElement().getId().equals(factor.getVariableId())) {
-                    // the sensitivity on a removed branch is 0, the sensitivity if the variable was a removed branch is 0
+                if (contingencyElement.getElement().getId().equals(functionBranchId)) {
+                    // the monitored branch is in contingency, the sensitivity value equals to zero and its post-contingency flow
+                    // equals to zero too.
                     sensiValue = 0d;
                     flowValue = 0d;
                     break;
                 }
                 double contingencySensitivity = p1.calculateSensi(contingenciesStates, contingencyElement.getContingencyIndex());
                 flowValue += contingencyElement.getAlphaForFunctionReference() * contingencySensitivity;
                 sensiValue +=  contingencyElement.getAlphaForSensitivityValue() * contingencySensitivity;
+                if (contingencyElement.getElement().getId().equals(factor.getVariableId())) {
+                    // the equipment responsible for the variable is indeed in contingency, the sensitivity value equals to zero.
+                    // No assumption about the reference flow on the monitored branch.
+                    sensiValue = 0d;
+                }
```

Comment:
It might also be worth it to compute the flows once per contingencyElements set and functionBranch (since it does not depend on the factor variable (pst)), but the gain is probably quite small.